src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -204,8 +207,16 @@ public RawAtlasSlicer(final Set<IsoCountry> countries,
         final CountryCodeProperties countryDetails = getCountryBoundaryMap()
                 .getCountryCodeISO3(location);
 
-        // Store the country code
-        tags.put(ISOCountryTag.KEY, countryDetails.getIso3CountryCode());
+        // Store the country code, enforce alphabetical order if there are multiple
+        if (countryDetails.inMultipleCountries())
+        {
+            tags.put(ISOCountryTag.KEY, Joiner.on(",").join(Sets.newTreeSet(Arrays.asList(
```

Comment:
Quick Questions: 

1. Have you seen country codes getting duplicated before or its just an extra piece of validation before we add the tag?
2. What kind of gains do we get from ordering the country codes? As adding to tree set from an ArrayList will still be  ```O(n * log(n))```.

Reply:
Great questions! 

1. No duplication. For synthetic nodes created for features that span the boundary, we will have two or more country codes. Same goes for polygons that are cut across a boundary. The new line that will connect them is on top of the country boundary and will have two country codes.
2. Key gain isn't performance, it's consistency and deterministic ordering. This ensures that running slicing will give us the same LineString with the same metadata each time. I figured this is pretty important in debugging various issues as they come up. Testing this, I've only seen a maximum of 3 country codes at an intersection where there are 3 countries, so this has pretty low performance impact.

Reply:
Thanks for the explanation. Having deterministic ordering makes more sense now. I kind of suspected that the country list won't be that long to have a performance impact but just wanted to be sure for my understanding. This makes much more sense  now